src/node_crypto.cc

```diff
@@ -3603,6 +3603,12 @@ bool PublicKeyCipher::Cipher(const char* key_pem,
   if (EVP_PKEY_CTX_set_rsa_padding(ctx, padding) <= 0)
     goto exit;
 
+  /* Reset md for signatures, we are doing raw RSA ops anyway */
+  if (EVP_PKEY_CTX_get_operation(ctx) == EVP_PKEY_OP_SIGN) {
+    if (EVP_PKEY_CTX_set_signature_md(ctx, nullptr) <= 0)
```

Comment:
Oops forgot to remove it, it is no longer needed. :) This piece was failing because PSS padding mode set `sha1` `md` and `rsa_pmeth.c` was ensuring that the input length matches the `md` output length. However, PSS does not really work with all these shit, so I replaced it with PKCS1, and PKCS1 does not set `md`.